lib/Conversion/CfToHandshake/CfToHandshake.cpp

```diff
@@ -1165,22 +1170,41 @@ ConvertCalls::matchAndRewrite(func::CallOp callOp, OpAdaptor adaptor,
   Operation *lookup = modOp.lookupSymbol(symbol);
   if (!lookup)
     return callOp->emitError() << "call references unknown function";
-  auto calledFuncOp = dyn_cast<handshake::FuncOp>(lookup);
-  if (!calledFuncOp)
-    return callOp->emitError() << "call does not reference a function";
-  TypeRange resultTypes = calledFuncOp.getFunctionType().getResults();
+  TypeRange tmp_resultTypes;
+  // check if the function is a handshake function
+  auto calledFuncOp = dyn_cast<handshake::FuncOp>(lookup);  
+  if (!calledFuncOp){
+    // if this is not the case, the function might have been not traversed yet during the conversion
+    auto _calledFuncOp = dyn_cast<func::FuncOp>(lookup);
+    if(!_calledFuncOp)
+      return callOp->emitError() << "call does not reference a function";
+    tmp_resultTypes = _calledFuncOp.getFunctionType().getResults();
+  } else {
+    tmp_resultTypes = calledFuncOp.getFunctionType().getResults();
+  }
+  std::vector<mlir::Type> customTypeVector;
```

Comment:
This is likely to be small so a `SmallVector` may avoid a heap allocation in the majority of cases.